src/coap_openssl.c

```diff
@@ -34,10 +41,21 @@ typedef struct coap_tls_context_t {
   BIO_METHOD *meth;
 } coap_tls_context_t;
 
+#define IS_PSK 0x1
+#define IS_PKI 0x2
+
+typedef struct sni_entry {
+  char *sni;
+  SSL_CTX *ctx;
+} sni_entry;
+
 typedef struct coap_openssl_context_t {
   coap_dtls_context_t dtls;
   coap_tls_context_t tls;
+  coap_dtls_pki_t setup_data;
   int psk_pki_enabled;
+  int sni_count;
```

Comment:
Can this be negative? Otherwise, it should be of an unsigned type as always.

Reply:
Fixed to be unsigned.